Build failure: lime3ds

## Steps To Reproduce

Steps to reproduce the behavior:

1. build *lime3ds*

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
error: builder for '/nix/store/1j8szpfw87bwzjh5cw4rxxwa863dkb6c-lime3ds-2119.1.drv' failed with exit code 2;
       last 25 log lines:
       > [ 81%] Building CXX object src/core/CMakeFiles/lime_core.dir/arm/dynarmic/arm_dynarmic_cp15.cpp.o
       > [ 82%] Building CXX object src/core/CMakeFiles/lime_core.dir/arm/dynarmic/arm_exclusive_monitor.cpp.o
       > [ 82%] Building CXX object src/core/CMakeFiles/lime_core.dir/arm/dynarmic/arm_tick_counts.cpp.o
       > In file included from include/boost/asio.hpp:69,
       >                  from /build/source/src/core/rpc/udp_server.cpp:6:
       > include/boost/asio/co_composed.hpp:849:13: error: 'async_operation' has not been declared
       > include/boost/asio/co_composed.hpp:850:24: error: 'Op' has not been declared
       > include/boost/asio/co_composed.hpp: In member function 'auto boost::asio::detail::co_composed_promise<Executors, Handler, Return>::await_transform(int&&)':
       > include/boost/asio/co_composed.hpp:862:19: error: expected ')' before '&&' token
       > include/boost/asio/co_composed.hpp:862:16: note: to match this '('
       > include/boost/asio/co_composed.hpp:915:7: error: 'Op' does not name a type
       > include/boost/asio/co_composed.hpp:925:34: error: 'Op' was not declared in this scope; did you mean 'op'?
       > include/boost/asio/co_composed.hpp:925:21: error: expected primary-expression before '{' token
       > include/boost/asio/co_composed.hpp:925:21: error: expected ';' before '{' token
       > include/boost/asio/co_composed.hpp:925:34: error: 'Op' does not name a type
       > include/boost/asio/co_composed.hpp:925:36: error: expected '>' before '&&' token
       > include/boost/asio/co_composed.hpp:925:36: error: expected '(' before '&&' token
       > include/boost/asio/co_composed.hpp:925:38: error: expected identifier before '>' token
       > include/boost/asio/co_composed.hpp:931:7: error: expected ')' before '}' token
       > At global scope:
       > cc1plus: note: unrecognized command-line option '-Wno-unused-command-line-argument' may have been intended to silence earlier diagnostics
       > make[2]: *** [src/core/CMakeFiles/lime_core.dir/build.make:3338: src/core/CMakeFiles/lime_core.dir/rpc/udp_server.cpp.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[1]: *** [CMakeFiles/Makefile2:1324: src/core/CMakeFiles/lime_core.dir/all] Error 2
       > make: *** [Makefile:146: all] Error 2
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
```
❯ nix-shell -p nix-info --run "nix-info -m"     
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.69, NixOS, 25.05 (Warbler), 25.05.20250102.6df2492`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/apfqvr9kddcl6fscjvb92p4xdrqzcwk0-source`
```

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@arthsmn 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/li/lime3ds/package.nix
+++ b/pkgs/by-name/li/lime3ds/package.nix
@@ -7,6 +7,7 @@
   cpp-jwt,
   doxygen,
   enet,
+  fetchpatch,
   fetchzip,
   fmt,
   ffmpeg_6-headless,
@@ -111,6 +112,14 @@ stdenv.mkDerivation (finalAttrs: {
     ++ optionals enableQtTranslations [ kdePackages.qttools ]
     ++ optionals enableCubeb [ cubeb ]
     ++ optional useDiscordRichPresence rapidjson;
+
+  patches = [
+    # Fix boost errors
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/Tatsh/tatsh-overlay/fa2f92b888f8c0aab70414ca560b823ffb33b122/games-emulation/lime3ds/files/lime3ds-0002-boost-fix.patch";
+      hash = "sha256-XJogqvQE7I5lVHtvQja0woVlO40blhFOqnoYftIQwJs=";
+    })
+  ];
 
   postPatch = ''
     # Fix file not found when looking in var/empty instead of opt

